game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -1752,19 +1753,45 @@ private void resetCanDieFromReachingMaxDamage() {
   }
 
   private void setIsSuicide(final String s) {
-    isSuicide = getBool(s);
+    setIsSuicide(getBool(s));
   }
 
   private void setIsSuicide(final Boolean s) {
```

Comment:
Would it maybe be more clear and direct to have a 'isSucicideOnAttack' property? It seems to get at that property, one has to be pretty clever to set a combination of values.

I wonder as well if simply adding those two properties would make it easier for this update to be backward compatible. I'm very concerned that the property is not directly from the XML but is instead derived, it seems like it is quite complex compared to other properties that are usually 1:1.

Reply:
Second, if 'isFirstStrike' is set, would this conflict and/or introduce a race condition for which property is actually set?
EG: isSuicide = true & isFirstStrike = false (or vice versa)

If this also gets reset, will that turn off firstStrike for units that have it? Are we guaranteed to have it be turned back on?

I'm feeling pretty convinced this property should be more independent of others, and allow 'isSuicide' to be more legacy and/or operate mostly as it used to.

Reply:
Not really following you here. So essentially what this does is add 2 new unit options `isSuicideOnAttack` and `isSuicideOnDefense` as often you don't want units to suicide in both cases (think I don't want missiles to suicide on defense, etc). This then deprecates `isSuicide` which for backwards compatibility now just sets these new properties. Eventually after this release and if all maps using `isSuicide` are updated then we could remove it all together.

Reply:
Essentially I was suggesting to leave the 'isSuicide' property unchanged and instead just add the new properties. It's more clear to me now that you're removing the old 'isSuicide' property.